p2p/transport/webtransport/transport.go

```diff
@@ -405,9 +404,10 @@ func (t *transport) Resolve(_ context.Context, maddr ma.Multiaddr) ([]ma.Multiad
 	return []ma.Multiaddr{beforeQuicMA.Encapsulate(quicComponent).Encapsulate(sniComponent).Encapsulate(afterQuicMA)}, nil
 }
 
+// AddCertHashes adds the current certificate hashes to a multiaddress.
+// If called before Listen, it's a no-op.
```

Comment:
Wouldn't returning an error or boolean here be better? Then callers, like the BasicHost could emit errors if they wanted to so that users would know they've done something wrong.

Not a strongly held view here, just seems like someone calling this before the certmanager has been initialized probably did so erroneously and so they should be informed of their error.